src/test/scala/SystemTests.scala

```diff
@@ -329,18 +258,19 @@ def histoToSvg(title: String, imgWidth: Int, imgHeight: Int, bins: List[Int], mi
 }
 
 
-def loadHisto() = {
-  val source = scala.io.Source.fromFile("src/test/full-testresult-GTIRB.csv").getLines().toList
-  val headers = source.head.split(",")
-
-  val res = headers.map(h => h -> mutable.ArrayBuffer[String]()).toMap[String, mutable.ArrayBuffer[String]]
+def loadHisto(): Unit = {
+  Using(Source.fromFile("src/test/full-testresult-GTIRB.csv")) { source =>
+    val sourceList = source.getLines().toList
+    val headers = sourceList.head.split(",")
 
-  source.tail.map(line => {
-    val cols = line.split(",")
-    headers.zip(cols).foreach((h,v) => res(h).append(v))
-  })
+    val res = headers.map(h => h -> mutable.ArrayBuffer[String]()).toMap[String, mutable.ArrayBuffer[String]]
 
-  val timeValues = res("verifyTime").map(_.toDouble)
-  val histo = histogram(50, Some(800.0, 1000.0))(timeValues.toSeq)
-  println(histoToSvg("test histogram", 500, 300, histo, 800.0, 1000.0))
+    sourceList.tail.foreach { line =>
+      val cols = line.split(",")
+      headers.zip(cols).foreach((h, v) => res(h).append(v))
+    }
+    val timeValues = res("verifyTime").map(_.toDouble)
+    val histo = histogram(50, Some(800.0, 1000.0))(timeValues.toSeq)
+    println(histoToSvg("test histogram", 500, 300, histo, 800.0, 1000.0))
```

Comment:
IMO SystemTests is a bit of an abuse of the unit test framework, especially the result aggregating stuff I added on. I'd like to be able to run the tests concurrently, and ideally we'd be able to factor out the part that gets all the test cases out of the file heirachy, the part that runs the tests, and the processing of the results to tables, csv etc., to reuse them for other tests. 

It would be nice to just concurrent map over the a list of TestConfig to run all the tests, and produce a result, (e.g. containing the timing information we use here), and then be able to reduce it however we want.

Scalatest can kind of parallelise by cases rather than suite if you use a different test suite parent class, but I didn't have much success with it and the result aggregation of course makes race conditions.

Reply:
I agree that the result aggregating stuff is a bit of an abuse of the framework, but it isn't straight forward to replace it so I left it for now.

Reply:
Yeah I think its fine to leave for now